rust_crate/src/main.rs

```diff
@@ -1,17 +1,37 @@
+use clap::Parser;
+
+#[derive(Parser, Debug)]
+struct Args {
+    dart_args: Vec<String>,
+    #[arg(short, long)]
+    protoc_path: Option<String>,
+}
+
 #[cfg(not(target_family = "wasm"))]
 fn main() {
     use std::env;
     use std::fs;
     use std::path;
+    use std::path::PathBuf;
     use std::process;
+    use std::str::FromStr;
 
-    // Install Protobuf compiler and get the path.
-    let home_path = home::home_dir().unwrap();
-    let out_path = home_path.join(".local").join("bin");
-    fs::create_dir_all(&out_path).unwrap();
-    env::set_var("OUT_DIR", out_path.to_str().unwrap());
-    let (protoc_binary_path, _) = protoc_prebuilt::init("25.2").unwrap();
-    let protoc_path = protoc_binary_path.parent().unwrap().to_path_buf();
+    let args = Args::parse();
+
+    let protoc_path;
+
+    if let Some(p) = args.protoc_path {
+        let provided = PathBuf::from_str(&p).unwrap();
```

Comment:
Is it worth to add a check for path existence here? Like:
```rust
if !provided.exists() {
// inform a error
}
```

Reply:
Yeah, I think it's a good idea to do it :)

Reply:
Done